react_app/src/components/GameContainer.js

```diff
@@ -1,53 +1,154 @@
+        title
+        upfrontCost
+        deferredCost
+
+        image
+        file
+        content
+        prerequisites {
+          id
+          title
+        }
+      }
+      category {
+        id
+        title
+      }
+      prerequisites {
+        id
+        title
+      }
+      unlockDelay
     }
   }
 `;
 
-const useStyles = makeStyles(theme => ({
+const styles = makeStyles(theme => ({
   root: {
     padding: theme.spacing()
   }
 }));
 
-export default function GameContainer(props) {
-  const classes = useStyles();
-
-  return (
-    <div className={classes.root}>
-      <Grid
-        container
-        spacing={3}
-        direction="row"
-        justify="space-around"
-        alignItems="center"
+class GameContainer extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      challengeDetail: null
+    };
+    this.openDetail = this.openDetail.bind(this);
+    this.closeDetail = this.closeDetail.bind(this);
+  }
+
+  openDetail(id) {
+    this.setState({ challengeDetail: id });
+  }
+  closeDetail() {
+    this.setState({ challengeDetail: null });
+  }
+
+  render() {
+    return (
+      <GameContext.Provider
+        value={{ open: this.openDetail, close: this.closeDetail }}
       >
-        <Query query={GET_ALL_CATEGORIES_QUERY} pollInterval={60000}>
-          {({ loading, error, data }) => {
-            if (loading) return <div>Fetching</div>;
-            if (error) return <div>Error</div>;
-
-            return data.allCategories.map(category => (
-              <Grid item lg={6}>
-                <CTFCategory key={category.id} category={category} />
-              </Grid>
-            ));
-          }}
-        </Query>
-      </Grid>
-    </div>
-  );
+        <div className={styles.root}>
+          <Grid
+            container
+            spacing={3}
+            direction="row"
+            justify="space-around"
+            alignItems="center"
+          >
+            <Query query={GET_ALL_CATEGORIES_QUERY} pollInterval={60000}>
+              {({ loading, error, data }) => {
+                if (loading) return <div>Fetching</div>;
+                if (error) return <div>Error</div>;
+
+                return data.allCategories.map(category => (
+                  <Grid item lg={6}>
+                    <CTFCategory
+                      key={category.id}
+                      category={category}
+                      detailView={this.openDetail}
+                    />
+                  </Grid>
+                ));
+              }}
+            </Query>
+          </Grid>
+          {this.state.challengeDetail != null && (
+            <Query
+              query={GET_CHALLENGE_DETAIL}
+              variables={{ id: this.state.challengeDetail }}
+            >
+              {({ loading, error, data }) => {
+                if (loading) return <div>"..."</div>;
+                if (error) return <div>"Error"</div>;
+
+                console.log(data);
```

Comment:
👀

Reply:
debugging line...

I have no idea how anyone debugs transpiled code.

Reply:
Chrome devtools dude - there are these things called "sourcemaps" that the dev tools can use to get back to the equivalent line in the un-transpiled code.

Reply:
Sokath, his eyes open